### PR TITLE
feat: Add support for base64 encoded images to be manually added to Images cache.

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Adds a method so base64 encoded images can be manually added to the Images cache.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Replace or remove this text.

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/cache/images.dart [hunk 1]
```diff
@@ -49,6 +49,14 @@ class Images {
     _assets[name] = _ImageAsset.fromImage(image);
   }
 
+  /// Transform the base64 encoded image into an [Image] and adds it into the
+  /// cache.
+  Future<void> addBase64EncodedImage(String name, String base64Data) async {
```
Comment (spydon) [inline packages/flame/lib/src/cache/images.dart:54]:
```suggestion
  Future<void> addFromBase64Data(String name, String base64Data) async {
```
Maybe? That it's an image should already be implicit from the cache name

**Review by spydon — APPROVED** [review]

**Comment by spydon** [comment]
@erickzanardo I just stumbled upon this method which is already in the `Images` class:
image
Doesn't that do the same thing, except with the missing dispose?

**Comment by erickzanardo** [comment]
@spydon not really, that method will fetch from cache if the image is already there, the method that I introduced will always force the cache to be updated.

**Comment by spydon** [comment]
> @spydon not really, that method will fetch from cache if the image is already there, the method that I introduced will always force the cache to be updated. 

Ah, alright, it seems like it should just have been a flag to this method instead of a new method then?

We should really standardize how the caches work for V2, it feels quite unstructured now.

**Comment by spydon** [comment]
> Yeah, agree on the flag, that was what I thought at first, but I preferred to follow the other method `add` that already exists to keep the consistency.

Makes sense, I'll put it in the V2 list that we should look through the cache classes :)

**Comment by erickzanardo** [comment]
Sounds good!!!!

**Pull request merged**